geth/api/backend.go

```diff
@@ -251,16 +253,27 @@ func (b *StatusBackend) getVerifiedAccount(password string) (*account.SelectedEx
 	return selectedAccount, nil
 }
 
-// ApproveSignRequest instructs backend to complete sending of a given transaction
-func (b *StatusBackend) ApproveSignRequest(id string, password string) sign.Result {
-	return b.pendingSignRequests.Approve(id, password, b.getVerifiedAccount)
+// ApproveSignRequest instructs backend to complete sending of a given transaction.
+// Empty values for gas or gasPrice will preserve values for these properties
+// as they were defined on the transaction initialization.
+func (b *StatusBackend) ApproveSignRequest(id, password string, gas, gasPrice int64) sign.Result {
+	args := sign.TxArgs{}
+	if gas > 0 {
```

Comment:
There doesn't seem to be any tests covering this logic. Someone can change this statement to e.g. `if gas == 0 {` and only an unrelated e2e jail test will fail:

```
    --- FAIL: TestJailRPCTestSuite/TestContractDeploymentLES (32.35s)
        jail_rpc_test.go:158: transaction queued and will be completed shortly, id: 2a4e895d-c4c2-4f4f-b421-86d6593cdeff
        Error Trace:    jail_rpc_test.go:163
                        signals.go:77
                        asm_amd64.s:573
                        cgocall.go:316
                        cgocall.go:194
                        asm_amd64.s:826
                        asm_amd64.s:673
                        cgocall.go:131
                        signals.go:41
                        events_sign.go:20
                        notifications.go:32
                        asm_amd64.s:2361
        Error:          Received unexpected error:
                        intrinsic gas too low
        Messages:       2a4e895d-c4c2-4f4f-b421-86d6593cdeff
```

If I change the line inside the `if` to `g := hexutil.Uint64(gas + 1)`, not a single test will fail. Maybe this warrants some unit tests?

Reply:
yep, you're right. Just added [TestPrepareTxArgs](https://github.com/status-im/status-go/pull/1000/commits/bde37f3ea54739e85281ceb09cfe629dba2c040e#diff-4b7f80343af3560f6214c875c1daee54R248) to cover that logic.